### DISCUSSION TITLE
Installation of Copilot for Visual Studio breaks literally every package in Visual Studio 2022 Community.

### DISCUSSION THREAD

**Discussion by Randname666** [community/community#17304] [Copilot Conversations] [answered]
After installing Copilot in Visual Studio 2022 Community, literally all packages, including critical ones that maintain basic Visual Studio functionalities, are in a broken state. 
When loading packages, such message pops up one after another:
```

The 'MicrosoftVisualStudio.Editor.Implementation.EditorPackage'package did not load correctly.

The problem may have been caused by a configuration change or by the installation of another extension. You can get more information by examining the file

'C:\Users\DELL\AppData\Roaming\Microsoft\VisualStudio\17.0_8d359511\ActivityLog.xml'.

Restarting Visual Studio could help resolve this issue.

Continue to show this error message?
```
The sample of ActivityLog.xml could be viewed at https://pastebin.com/6TPAeqdg .

Also this caused Visual Studio unable to quit after opening a solution file, with another error message:
```
Expected 1 export(s) with contract name "Microsoft.Internal.VisualStudio.PlatformUI.ISolutionAttachedCollectionService" but found 0 after applying applicable constraints.
```

**Comment by Xteuk** [comment]
Same here yesterday (2022/06/02), on a brand new installation of VS 2022 Professionnal (french version) with "C++ Desktop" and ".Net Desktop".

Every single command is out of order. Only fix found to totally remove VS and reinstall.

**Comment by Simulacrux** [comment]
Same here installed it onto a brand new updated Visual Studio Enterprise 2022 (Version 17.2.3) with several Packages installed (Like C#, .Net, C++, Asp.Net...) 
If you click on "Create new project" no project templates are shown and something like this error appears:
```
---------------------------
Microsoft Visual Studio
---------------------------
The 'Background long idle package' package did not load correctly.

The problem may have been caused by a configuration change or by the installation of another extension. You can get more information by examining the file 'D:\Users\dholzapfel\AppData\Roaming\Microsoft\VisualStudio\17.0_e896e942\ActivityLog.xml'.

Restarting Visual Studio could help resolve this issue.

Continue to show this error message?
---------------------------
Yes   No   
---------------------------

```
![image](https://user-images.githubusercontent.com/12858171/171750223-5be28a8a-492a-40d6-b47a-b52b8822fbb9.png)

And if I try to open an older solution nearly every menu command is prompting only this error:
```
---------------------------
Microsoft Visual Studio
---------------------------
Expected 1 export(s) with contract name "Microsoft.Internal.VisualStudio.PlatformUI.ISolutionAttachedCollectionService" but found 0 after applying applicable constraints.
---------------------------
OK   
---------------------------
```
![image](https://user-images.githubusercontent.com/12858171/171750389-5a3f11e6-979c-4a5b-b4a5-db541be2f6dc.png)

Will now try to clear this mess up without reinstall, lets see which reset is maybe working 👨‍🏭

**Update**
- Uninstalling the Extension on a Developer Command Prompt with `vsixinstaller /u:GitHub.Copilot` did not help
- Starting the VisualStudio with `devenv /SafeMode` worked, but I could not solve anything from within
- Calling a `devenv /ResetSettings` did not bring me further
- Calling a `devenv /ResetUserData` now brought me back a "working" VisualStudio but lost all Extensions and Settings from before, so only half a good solution

**Reply by Simulacrux** [reply]
After burning my fingers the first time I tried it again on the same system after the devenv /ResetUserData, but just installing Github Copilot Extension broke it right away. Another Reset fixed it again. 
Just one other info, it is a AWS Workspace virtualization and I think a Windows 2016 as OS. 
Now I roll some dice and figure out, if I will break also my other Visual Studio installation with testing.

**Reply by Randname666** [reply]
> Same here installed it onto a brand new updated Visual Studio Enterprise 2022 (Version 17.2.3) with several Packages installed (Like C#, .Net, C++, Asp.Net...) If you click on "Create new project" no project templates are shown and something like this error appears:

In fact, this also happens with a Visual Studio installation with no workloads installed but only the editor and IntelliCode (as Copilot relies on that).

**Reply by guifran001** [reply]
In happens with a vanilla version of Visual Studio Professional as well. Just installing the copilot extension is causing the symptoms described above...

**Comment by guifran001** [comment]
I got: `MEF Service Broker Package`, `Whats New Package` and `Background long idle package` that did not load correctly after installing CoPilot.

Afterward, if I return in Manage Extension, copilot is not installed and if I try to download it, it prompts an error stating that Intellicode is not installed, but when going to Options, it is activated.

**Comment by Randname666** [comment, marked as answer]
Okay, I think I have the problem fixed. Turned out it's caused by some orphaned old Visual Studio packages that mess up the vsix installer.
If you have upgraded to Visual Studio 2022 from older version of have other version of Visual Studio installed at the same time, then you have to clean them off first via [uninstaller](https://github.com/microsoft/VisualStudioUninstaller), then reinstall Visual Studio 2022.

**Reply by Xteuk** [reply]
It is true that I have multiple versions of Visual Studio installed on this computer.
(can't remove them, though)

Thank for finding that.

**Comment by epsilon-sagittarii** [comment]
New PC, clean installation of Windows and Visual Studio Community and facing same problem:

<img width="316" alt="image" src="https://github.com/community/community/assets/60514256/f4085a82-09b4-45e0-bc86-18640bc549de">

**Comment by joshfeimster** [comment]
Same issue -- just started on 11/30. I have had no issues for months. The only thing I added was Copilot Chat.

![image](https://github.com/community/community/assets/25965069/1073908a-8558-4c14-bb5d-e5f37ee0f7a1)
